src/sentry/api/endpoints/project_details.py

```diff
@@ -771,6 +778,36 @@ def delete(self, request: Request, project) -> Response:
 
         return Response(status=204)
 
+    @staticmethod
+    def _is_uniform_sampling_rule(rule):
+        # Currently UI sends top level condition op as always 'and' so asserting this here to
+        # guarantee that it will always have an `inner` key
+        assert rule["condition"]["op"] == "and"
+        # Matching the uniform sampling rule check on UI
+        return len(rule["condition"]["inner"]) == 0 and rule["type"] == "trace"
```

Comment:
Maybe we can rephrase the comment here, it is not about the UI.
A uniform sampling rule must be an 'and' with no rules. 
Note that an 'or' with no rules will not match anything. 
Think about an `and` as starting as `True` and being set to `False` by the first failure, and an `or` as
starting as `False` and being set to `True` by the first rule that passes.

Also not sure why are you checking for rule['type'] == 'trace' (but it may be a valid reason)

Reply:
>Maybe we can rephrase the comment here, it is not about the UI.
A uniform sampling rule must be an 'and' with no rules.
Note that an 'or' with no rules will not match anything.

Agreed. Will update comment 

>Also not sure why are you checking for rule['type'] == 'trace' (but it may be a valid reason)

Because we are currently not supporting creating uniform rules for transactions, just traces